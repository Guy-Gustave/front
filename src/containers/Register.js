/* eslint-disable */
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import registration from '../apiRequests/regRequest';
import RegisterStyles from '../styles/RegisterStyles.module.css';

const Register = props => {
  const dispatch = useDispatch();
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  // const [username, setUsername] = useState('');
  const [errorMsg, setErrorMsg] = useState('');

  const handleChange = event => {
    if (event.target.name === 'name') {
      setName(event.target.value);
    } else if (event.target.name === 'email') {
      setEmail(event.target.value);
    } else if (event.target.name === 'username') {
      setUsername(event.target.value);
    } else {
      setPassword(event.target.value);
    }
  };

  const handleRegister = () => {
    const { history } = props;
    history.push('/airlinesList');
  };

  const handleSubmit = event => {
    event.preventDefault();
    dispatch(registration({
      name,
      email,
      password,
    }));
    if (email || password !== '') {
      setErrorMsg('');
      handleRegister();
    } else {
      setErrorMsg('Please, enter valid credentials!');
    }
  };

  const validateForm = () => {
    const x = document.forms.name.value;
    if (x === '') {
      // eslint-disable-next-line
      alert('All fields must be filled out');
      return false;
    }
    return true;
  };

  return (
    <div data-testid="container">
      <form onSubmit={validateForm} className={RegisterStyles.registerForm} data-testid="form">
        <Link className={RegisterStyles.homeLink} to="/">
          <button type="button" className={`${RegisterStyles.homeBtn} btn`}>Home</button>
        </Link>
        <label htmlFor="name" className="text-left">
          Name:
          <br />
          <input type="text" name="name" id="name" placeholder="Enter your name" onChange={handleChange} value={name.name} />
        </label>
        <br />
        {/* <label htmlFor="username" className="text-left">
          Username:
          <br />
          <input type="text" name="username" id="username" placeholder="Enter your username" onChange={handleChange} value={username.username} />
        </label> */}
        <br />
        <label htmlFor="email" className="text-left">
          Email:
          <br />
          <input type="text" name="email" id="email" placeholder="Enter your email" onChange={handleChange} value={email.email} />
        </label>
        <br />
        <label htmlFor="password" className="text-left">
          Password:
          <br />
          <input type="password" name="password" id="password" placeholder="Enter your password" onChange={handleChange} value={password.password} />
        </label>
        <br />
        <button type="submit" className={`${RegisterStyles.btn} btn`} onClick={handleSubmit}>Register</button>
        <div>
          <span>Already have an account? </span>
          <Link to="/login">Login</Link>
        </div>
        {errorMsg === '' ? '' : <h3 className="text-danger">{errorMsg}</h3>}
      </form>
    </div>
  );
};

Register.propTypes = {
  history: PropTypes.instanceOf(Object).isRequired,
};

export default Register;
