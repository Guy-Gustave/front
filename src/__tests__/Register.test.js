import React from 'react';
import { render } from '@testing-library/react';
// import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Register from '../containers/Register';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Register', () => {
  test('renders Register component', () => {
    const { queryAllByTestId } = render(
      (
        <Provider store={store}>
          <Router>
            <Route path="/register" component={Register} />
          </Router>
        </Provider>
      ),
    );
    const container = queryAllByTestId('container');
    expect(container).toBeTruthy();
  });

  test('it should have a Register form', () => {
    const { queryAllByTestId } = render(
      (
        <Provider store={store}>
          <Router>
            <Route path="/register" component={Register} />
          </Router>
        </Provider>
      ),
    );
    const form = queryAllByTestId('form');
    expect(form).toBeTruthy();
  });
});
