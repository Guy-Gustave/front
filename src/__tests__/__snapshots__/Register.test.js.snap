// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register matches Register snapshot 1`] = `
<div
  data-testid="container"
>
  <form
    className="registerForm"
    data-testid="form"
    onSubmit={[Function]}
  >
    <a
      className="homeLink"
      href="/"
      onClick={[Function]}
    >
      <button
        className="homeBtn btn"
        type="button"
      >
        Home
      </button>
    </a>
    <label
      className="text-left"
      htmlFor="name"
    >
      Name:
      <br />
      <input
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Enter your name"
        type="text"
      />
    </label>
    <br />
    <label
      className="text-left"
      htmlFor="email"
    >
      Email:
      <br />
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        type="text"
      />
    </label>
    <br />
    <label
      className="text-left"
      htmlFor="password"
    >
      Password:
      <br />
      <input
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
      />
    </label>
    <br />
    <label
      className="text-left"
      htmlFor="password_confirmation"
    >
      password_confirmation:
      <br />
      <input
        id="password_confirmation"
        name="password_confirmation"
        onChange={[Function]}
        placeholder="Enter your password_confirmation"
        type="password"
      />
    </label>
    <br />
    <button
      className="btn btn"
      onClick={[Function]}
      type="submit"
    >
       Register
    </button>
    <div>
      <span>
        Already have an account? 
      </span>
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
    
  </form>
</div>
`;
